lua/telescope/utils.lua

```diff
@@ -141,9 +141,14 @@ end
 function utils.buf_delete(bufnr)
   if bufnr == nil then return end
 
+  -- Suppress the buffer deleted message for those with &report<2
+  start_report, vim.o.report = vim.o.report, 2
```

Comment:
It seems like what I see is `plenary`'s `context_manager` would be useful here but I don't see one implemented for setting-and-then-resetting a vim option. Maybe that's generically useful to add there?

Reply:
Can we only set it if it's not a good value?

And yeah, it could be used with the context manager. It's not a bad idea at all.

Reply:
Done.